### PR TITLE
vala: Cover 'find_library' usage and be more explicit for default install directories in docs

### PR DESCRIPTION
**Pull request by arteymix**

### CODE REVIEW

File: docs/markdown/Vala.md [hunk 1]
```diff
@@ -36,9 +36,21 @@ foo_dep = dependency('foo') # 'foo.vapi' will be resolved in './vapi/foo.vapi'
 executable('app', 'app.vala', dependencies: [glib_dep, gobject_dep, foo_dep])
 ```
 
+In this case, make sure that the VAPI name correspond to the pkg-config file.
```
Comment (pitti) [inline docs/markdown/Vala.md]: corresponds

File: docs/markdown/Vala.md [hunk 2]
```diff
@@ -36,9 +36,21 @@ foo_dep = dependency('foo') # 'foo.vapi' will be resolved in './vapi/foo.vapi'
...
+foo_vapi = meson.get_compiler('vala').find_library('foo', dirs: join_paths(meson.current_source_dir(), 'vapi')
+foo_dep = declare_dependency(dependencies: [foo_lib, foo_vapi])
+
+executable('app', 'app.vala', dependencies: [glib_dep, gobject_dep, foo_dep])
+```
+
 ## Custom output names
 
-If a library target is used, Meson automatically output the C header and the VAPI. They can be renamed by setting the `vala_header` and `vala_vapi` arguments respectively. In this case, the second and third elements of the `install_dir` array indicate the destination with `true` to indicate default directories.
+If a library target is used, Meson automatically output the C header and the VAPI. They can be renamed by setting the `vala_header` and `vala_vapi` arguments respectively. In this case, the second and third elements of the `install_dir` array indicate the destination with `true` to indicate default directories (i.e. `include` and `share/vala/vapi`).
```
Comment (pitti) [inline docs/markdown/Vala.md]: outputs

Comment (pitti) [inline docs/markdown/Vala.md]: and also "second and third element"

File: docs/markdown/Vala.md [hunk 3]
```diff
@@ -36,9 +36,21 @@ foo_dep = dependency('foo') # 'foo.vapi' will be resolved in './vapi/foo.vapi'
 executable('app', 'app.vala', dependencies: [glib_dep, gobject_dep, foo_dep])
 ```
 
+In this case, make sure that the VAPI name correspond to the pkg-config file. 
+
+If no pkg-config file is provided, you must use `find_library`. Using`declare_dependency` is cleaner because it does not require passing both dependency objects to the target.
+
+```meson
+foo_lib = meson.get_compiler('vala').find_library('foo') # assuming libfoo.so is installed
```
Comment (nirbheek) [inline docs/markdown/Vala.md]: This should be `meson.get_compiler('c')` to find `libfoo.so`.

File: docs/markdown/Vala.md [hunk 4]
```diff
@@ -36,9 +36,21 @@ foo_dep = dependency('foo') # 'foo.vapi' will be resolved in './vapi/foo.vapi'
 executable('app', 'app.vala', dependencies: [glib_dep, gobject_dep, foo_dep])
 ```
 
+In this case, make sure that the VAPI name correspond to the pkg-config file. 
+
+If no pkg-config file is provided, you must use `find_library`. Using`declare_dependency` is cleaner because it does not require passing both dependency objects to the target.
```
Comment (nirbheek) [inline docs/markdown/Vala.md]: *If no pkg-config file is provided and no vapi file is available on the system.

Comment (arteymix) [inline docs/markdown/Vala.md]: It's in the custom VAPI section, so I'm sort of assuming that the user will be providing it. I can reword this to cover the case of a VAPI without a corresponding pkg-config file like `posix.vapi`.

File: docs/markdown/Vala.md (hunk 3 above)
Comment (arteymix) [inline docs/markdown/Vala.md]: Effectively, I'll correct that.

File: docs/markdown/Vala.md (hunk 2 above)
Comment (arteymix) [inline docs/markdown/Vala.md]: We're talking about two elements here. I'll fix the word "output" nevertheless.
